### PR TITLE
Feature/integrated integration tests

### PR DESCRIPTION
**Pull request by prescod**
Add VCR to our test suite.

Invoking pytest like this:

```pytest --org dev ```

Will overwrite VCR cassettes.

Invoking it without `--org` will use cassettes from git.

cumulusci/tests/pytest_plugins/pytest_sf_orgconnect.py is a plugin with minor changes from the old integration testing plugin. It doesn't know or care about VCR, but it knows about Salesforce orgs.

cumulusci/tests/pytest_plugins/pytest_sf_vcr.py is a new plugin which adapts VCR to the Salesforce context (scrubbing OrgIDs, SessionIDs, etc.)

The output of pytest_sf_vcr is visible in the cassette in this PR.

### CODE REVIEW

File: cumulusci/conftest.py [hunk 1]
```diff
@@ -51,3 +53,22 @@ def _make_response(status):
         return MockHttpResponse(status)
 
     return _make_response
+
+
+@fixture(scope="session")
+def fallback_orgconfig(request):
```
Comment (davisagli) [inline cumulusci/conftest.py:59]: Is depending on the request here important for some reason?

File: cumulusci/tests/pytest_plugins/pytest_sf_orgconnect.py [hunk 2]
```diff
@@ -24,15 +26,19 @@ def project_config(runtime):
...
     """
-    org_name = request.config.getoption("--org")
-    org_name, org_config = runtime.get_org(org_name)
-    org_config.refresh_oauth_token(runtime.keychain)
+    org_name = sf_pytest_orgname(request)
+    if org_name:
+        org_name, org_config = runtime.get_org(org_name)
+        org_config.refresh_oauth_token(runtime.keychain)
+    else:
+        return fallback_orgconfig()
```
Comment (davisagli) [inline cumulusci/tests/pytest_plugins/pytest_sf_orgconnect.py:40]: Maybe worth a comment about how fallback_orgconfig is defined differently for integration tests?

File: cumulusci/tests/pytest_plugins/pytest_sf_vcr.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        request.body = re.sub(
+            "<sessionId>.*</sessionId>",
+            "<sessionId>**Elided**</sessionId>",
+            request.body.decode(),
+        ).encode()
+    request.uri = re.sub(
+        "//.*.my.salesforce.com", "//orgname.salesforce.com", request.uri
+    )
+    request.uri = re.sub(
+        "//cs.*.salesforce.com/", "//podname.salesforce.com/", request.uri
```
Comment (davisagli) [inline cumulusci/tests/pytest_plugins/pytest_sf_vcr.py:17]: Note that the cs prefix is for sandbox pods, and other prefixes are used for production pods.

File: cumulusci/tests/pytest_plugins/pytest_sf_vcr.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    )
+    request.uri = re.sub("Organization/00.*", "Organization/ORGID", request.uri)
```
Comment (davisagli) [inline cumulusci/tests/pytest_plugins/pytest_sf_vcr.py:19]: What about matching something like `00D[\w\d]{15,18}` to catch all mentions of org ids?

File: cumulusci/tests/pytest_plugins/pytest_sf_vcr.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    return request
+
+
+# junk_headers = ["Public-Key-Pins-Report-Only", ]
+def sf_before_record_response(response):
+    # for header in junk_headers:
+    #     if response["headers"].get(header):
+    #         del response["headers"][header]
+    response["headers"] = {"Response-Headers": "Elided"}
+    return response
```
Comment (davisagli) [inline cumulusci/tests/pytest_plugins/pytest_sf_vcr.py:38]: remove commented code

File: cumulusci/tests/pytest_plugins/pytest_sf_vcr.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    summary1 = (r1.method, r1.uri, r1.body)
+    summary2 = (r2.method, r2.uri, r2.body)
+    print(summary1, summary2, summary1 == summary2)
+    assert summary1 == summary2
+
+
+def salesforce_vcr(vcr):
+    vcr.register_matcher("Salesforce Matcher", salesforce_matcher)
+    vcr.match_on = ["Salesforce Matcher"]
+    return vcr
```
Comment (davisagli) [inline cumulusci/tests/pytest_plugins/pytest_sf_vcr.py:75]: Could you add some comments about the ways that we are customizing vcr?

File: integration_tests/README.rst [hunk 7]
```diff
@@ -6,9 +6,10 @@ a different directory.
 
 You can invoke these tests with the command:
 
-    pytest tests/integration --org <orgname>
+    pytest integration_tests/integration --org <orgname>
 
-You can leave out the ``--org`` option if you have an org defaulted.
+--org should be a garbage org that you do not care about because it may be
+      changed.
```
Comment (davisagli) [inline integration_tests/README.rst:12]: Should we add an assertion somewhere that it is a scratch org?

File: cumulusci/tests/pytest_plugins/pytest_sf_vcr.py (hunk 3 above)
Comment (prescod) [reply cumulusci/tests/pytest_plugins/pytest_sf_vcr.py:17]: I'll jut open it up to everything with digits in the domain before the salesforce.com. ".my" URLs will have already been transformed to NOT have digits in the line before.

File: cumulusci/tests/pytest_plugins/pytest_sf_orgconnect.py (hunk 2 above)
Comment (prescod) [reply cumulusci/tests/pytest_plugins/pytest_sf_orgconnect.py:46]: https://github.com/SFDO-Tooling/CumulusCI/pull/1816/commits/e1bdf07bc3899f6d19b51a13bf5c997503dc9c1d#diff-20c920c476d998d34927d826ade54c1dR41

File: cumulusci/tests/pytest_plugins/pytest_sf_vcr.py (hunk 5 above)
Comment (prescod) [reply cumulusci/tests/pytest_plugins/pytest_sf_vcr.py:37]: https://github.com/SFDO-Tooling/CumulusCI/pull/1816/commits/e1bdf07bc3899f6d19b51a13bf5c997503dc9c1d#diff-9b5998118d11aec9e8e2fb607f028f78L34

File: cumulusci/tests/pytest_plugins/pytest_sf_vcr.py (hunk 6 above)
Comment (prescod) [reply cumulusci/tests/pytest_plugins/pytest_sf_vcr.py:76]: https://github.com/SFDO-Tooling/CumulusCI/pull/1816/commits/e1bdf07bc3899f6d19b51a13bf5c997503dc9c1d#diff-9b5998118d11aec9e8e2fb607f028f78R1

And:

https://github.com/SFDO-Tooling/CumulusCI/pull/1816/commits/e1bdf07bc3899f6d19b51a13bf5c997503dc9c1d#diff-9b5998118d11aec9e8e2fb607f028f78R79

File: integration_tests/README.rst (hunk 7 above)
Comment (prescod) [reply integration_tests/README.rst:12]: https://github.com/SFDO-Tooling/CumulusCI/pull/1816/commits/e1bdf07bc3899f6d19b51a13bf5c997503dc9c1d#diff-20c920c476d998d34927d826ade54c1dR38

File: cumulusci/conftest.py (hunk 1 above)
Comment (prescod) [reply cumulusci/conftest.py]: No, I've taken it out.

https://github.com/SFDO-Tooling/CumulusCI/pull/1816/files#diff-6fc3836ef3692ac53afe24241373e54dR59

File: cumulusci/tests/pytest_plugins/pytest_sf_vcr.py (hunk 3 above)
Comment (prescod) [reply cumulusci/tests/pytest_plugins/pytest_sf_vcr.py]: https://github.com/SFDO-Tooling/CumulusCI/pull/1816/files#diff-9b5998118d11aec9e8e2fb607f028f78R25

File: cumulusci/tests/pytest_plugins/pytest_sf_vcr.py (hunk 4 above)
Comment (prescod) [reply cumulusci/tests/pytest_plugins/pytest_sf_vcr.py]: https://github.com/SFDO-Tooling/CumulusCI/pull/1816/files#diff-9b5998118d11aec9e8e2fb607f028f78R27

**Pull request merged**